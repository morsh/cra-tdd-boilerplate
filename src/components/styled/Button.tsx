import styled, { css } from 'styled-components';

interface Props {
  primary?: boolean;
  ['data-hook']: string;
}

export const Button = styled.button.attrs<Props, Props>(props => ({
  ['data-hook']: props['data-hook'] || 'styled-button'
}))`
  /* This renders the buttons above... Edit me! */
  display: inline-block;
  border-radius: 3px;
  padding: 0.5rem 0;
  margin: 0.5rem 1rem;
  width: 11rem;
  background: transparent;
  color: white;
  border: 2px solid white;

  /* The GitHub button is a primary button
   * edit this to target it specifically! */
  ${props => props.primary && css`
    background: white;
    color: black;
  `}
`;
