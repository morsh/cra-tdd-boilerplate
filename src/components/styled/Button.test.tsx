import { render } from '@testing-library/react';
import 'jest-styled-components';
import { Button } from './Button';

describe('NavLink', () => {
  it('should be in document', () => {
    const { container } = render(<Button />);
    expect(container.querySelector('button')).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = render(<Button />);
    expect(container.querySelector('button')).toMatchSnapshot();
  });

  it('button doesnt render as primary', () => {
    const { container } = render(<Button primary />);
    expect(container.querySelector('button')).toHaveStyleRule('background', 'white');
    expect(container.querySelector('button')).toHaveStyleRule('color', 'black');
  });
});
