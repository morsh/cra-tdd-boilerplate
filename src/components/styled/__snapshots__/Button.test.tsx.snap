// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavLink should match snapshot 1`] = `
.c0 {
  display: inline-block;
  border-radius: 3px;
  padding: 0.5rem 0;
  margin: 0.5rem 1rem;
  width: 11rem;
  background: transparent;
  color: white;
  border: 2px solid white;
}

<button
  class="c0"
  data-hook="styled-button"
/>
`;
